lib/primitives/mtx.js

```diff
@@ -868,6 +868,51 @@ class MTX extends TX {
     return null;
   }
 
+  /**
+   * Sign custom P2WSH input.
+   * @param {Number} index - Index of input being signed.
+   * @param {KeyRing[]} rings - with private key.
+   * Sorted by desired signature order.
+   * @param {Script} redeemScript - its hash is in spending output.
+   * @param {Buffer|Number[]} witnessScript - signatures template.
+   * witnessScript is the SegWit equivalent of scriptSig.
+   * Put an empty buffer as placeholder for each signature.
+   * @param {SighashType} type
+   * @param {Coin|Output|Number} coin or output or value.
```

Comment:
I don't love how polymorphic the `coin` parameter is here.

Reply:
Since `mtx.sign` assumes the coinview is populated, I think that you could just remove the need for the `coin` parameter completely and work under the same assumption.

`mtx.addCoin` will populate both the coinview and the mtx.input, which is what you are using to test this method below.

https://github.com/bcoin-org/bcoin/blob/67d444b5ec45e45147bc780573494334c1ef3019/lib/primitives/mtx.js#L1053